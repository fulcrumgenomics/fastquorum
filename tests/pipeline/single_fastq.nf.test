nextflow_pipeline {

    name "Test workflow - Single FASTQ"
    script "main.nf"
    tag "single_fastq"
    tag "pipeline"

    test("Single FASTQ - R&D mode") {

        when {
            params {
                input      = 'https://raw.githubusercontent.com/nf-core/test-datasets/fastquorum/testdata/samplesheets/samplesheet.single_fastq.csv'
                fasta      = 'https://github.com/nf-core/test-datasets/raw/fastquorum/references/chr17.fa'
                outdir     = "$outputDir"
                mode       = "rd"
                duplex_seq = false
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("single_fastq_rd_software_versions") },
                { assert workflow.trace.succeeded().size() == 11 },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_one_fastq/SRR6109255_one_fastq.cons.filtered.bam").exists() },
                { assert snapshot(
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_one_fastq/SRR6109255_one_fastq.grouped-family-sizes.txt"),
                ).match("single_fastq_rd") }
            )
        }

    }

    test("Single FASTQ - high throughput mode") {

        when {
            params {
                input      = 'https://raw.githubusercontent.com/nf-core/test-datasets/fastquorum/testdata/samplesheets/samplesheet.single_fastq.csv'
                fasta      = 'https://github.com/nf-core/test-datasets/raw/fastquorum/references/chr17.fa'
                outdir     = "$outputDir"
                mode       = "ht"
                duplex_seq = false
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("single_fastq_ht_software_versions") },
                { assert workflow.trace.succeeded().size() == 10 },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/consensus_filtering/filtered/SRR6109255_one_fastq/SRR6109255_one_fastq.cons.filtered.bam").exists() },
                { assert snapshot(
                    path("$outputDir/grouping/groupreadsbyumi/SRR6109255_one_fastq/SRR6109255_one_fastq.grouped-family-sizes.txt"),
                ).match("single_fastq_ht") }
            )
        }

    }

}
