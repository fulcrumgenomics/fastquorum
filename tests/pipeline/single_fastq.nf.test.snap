{
    "single_fastq_rd": {
        "content": [
            "SRR6109255_one_fastq.grouped-family-sizes.txt:md5,8ee886d2141217391ed659819c0f37c0"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T07:33:11.408542"
    },
    "single_fastq_ht_software_versions": {
        "content": [
            "{ALIGN_CONSENSUS_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, ALIGN_RAW_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, CALLANDFILTERMOLECULARCONSENSUSREADS={fgbio=2.0.2}, FASTQC={fastqc=0.12.1}, FASTQTOBAM={fgbio=2.0.2}, GROUPREADSBYUMI={fgbio=2.0.2}, Workflow={nf-core/fastquorum=v1.0.0}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T16:28:01.840894"
    },
    "single_fastq_rd_software_versions": {
        "content": [
            "{ALIGN_CONSENSUS_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, ALIGN_RAW_BAM={bwa=0.7.17-r1188, fgbio=2.0.2, samtools=1.16.1}, CALLMOLECULARCONSENSUSREADS={fgbio=2.0.2}, FASTQC={fastqc=0.12.1}, FASTQTOBAM={fgbio=2.0.2}, FILTERCONSENSUSREADS={fgbio=2.0.2}, GROUPREADSBYUMI={fgbio=2.0.2}, Workflow={nf-core/fastquorum=v1.0.0}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T16:21:29.23891"
    },
    "single_fastq_ht": {
        "content": [
            "SRR6109255_one_fastq.grouped-family-sizes.txt:md5,8ee886d2141217391ed659819c0f37c0"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-20T07:39:38.166122"
    }
}